src/lang-promql/promql.ts

```diff
@@ -0,0 +1,28 @@
+import { LezerSyntax } from "@codemirror/next/syntax";
+import { parser } from "lezer-promql";
+import { styleTags } from "@codemirror/next/highlight";
+import { Extension } from "@codemirror/next/state";
+
+export const promQLSyntax = new LezerSyntax(parser.withProps(
+  styleTags({
+    LineComment: "comment",
+    LabelName: "name",
+    StringLiteral: "string",
+    NumberLiteral: "number",
+    Duration: "number",
+    "abs absent absent_over_time avg_over_time ceil changes clamp_max clamp_min count_over_time days_in_month day_of_month day_of_week delta deriv exp floor histogram_quantile holt_winters hour idelta increase irate label_replace label_join ln log10 log2 max_over_time min_over_time minute month predict_linear quantile_over_time rate resets round scalar sort sort_desc sqrt stddev_over_time stdvar_over_time sum_over_time time timestamp vector year": "keyword",
+    "avg bottomk count count_values max min quantile stddev stdvar sum topk": "keyword definition",
+    "group_left group_right by without bool offset on and or unless": "operator",
+    BinOp: "operator",
+    MatchOp: "operator",
+    UnaryOp: "operator",
+    "( )": "paren",
+    "[ ]": "squareBracket",
+    "{ }": "brace",
+  })
+))
```

Comment:
I also added the following, though I think it might not be needed:

```
  {
    languageData: {
      closeBrackets: { brackets: ['(', '[', '{', "'", '"', '`'] },
      commentTokens: { line: '#' },
    },
  }
```

Reply:
yeah I don't think it's needed.

Reply:
Also not the comment token? For automatic toggling of line comments? See https://codemirror.net/6/docs/ref/#comment

Reply:
ah Sorry I didn't read correctly. Yes let's add it, it won't hurt if people don't want to use the closing feature

Reply:
I don't see the effect of adding that, but maybe I missed a configuration somewhere. In any case I added it